client/services/FeedbackGeneratorService.js

```diff
@@ -55,6 +56,9 @@ tie.factory('FeedbackGeneratorService', [
 
     return {
       getFeedback: function(prompt, codeEvalResult) {
+        // TODO(eyurko): This is getting long enough that we should
```

Comment:
Mmm. How about pulling just the buggy function stuff out into a separate function now? Otherwise you have a giant block of code here -- and then the next PR will contain two giant blocks of code that are largely the same but hard to compare.